clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -845,6 +882,30 @@ private void handleListOffsetResponse(Map<TopicPartition, Long> timestampsToSear
         return parsedRecords;
     }
 
+    private PriorityQueue<FetchResponse.AbortedTransaction> getAbortedTransactions(FetchResponse.PartitionData partition) {
+        PriorityQueue<FetchResponse.AbortedTransaction> abortedTransactions = null;
+        if (partition.abortedTransactions != null && 0 < partition.abortedTransactions.size()) {
+            abortedTransactions = new PriorityQueue<>(
+                    partition.abortedTransactions.size(),
+                    new Comparator<FetchResponse.AbortedTransaction>() {
+                        @Override
+                        public int compare(FetchResponse.AbortedTransaction o1, FetchResponse.AbortedTransaction o2) {
```

Comment:
This is a good point. The document doesn't specify it. I think we have to wait for the implementation, but intuitively, the entries in the abort index need not be sorted by first offset since transactions can start and finish at any time. We either sort it on the broker after retrieval, or on the client. I think it makes more sense to sort on the client to keep the load on the server lower.